### PR TITLE
Update bundle

### PR DESCRIPTION
**Pull request by meowsus**
Closes #370

### CODE REVIEW

**Comment by meowsus** [comment]
@grosser is this what you're looking for?

**Comment by grosser** [comment]
yeah :D
once this is green we can bring back the `rm Gemfile.lock` so we actually test with the latest dependencies that end-users would be using :)

**Comment by meowsus** [comment]
@grosser I'm happy to look into the builds that are failing for Ruby v2.3.3 and v2.4.0, but I'm not sure how you want to resolve the dependency issues with v2.1.9 and v2.2.6. It looks like `public_suffix-4.0.1 requires ruby version >= 2.3` :|

**Comment by grosser** [comment]
Can kick out 2.1 and 2.2 since they are dead anyway 🥳

**Comment by meowsus** [comment]
@grosser sounds good!

**Comment by meowsus** [comment]
I took a few liberties with this branch, mostly because it was difficult to tell which adapter was actually causing these tests to fail. I kept my commits atomic, so let me know if any don't agree with you.

The failure was stemming from the nokogumbo adapter. Apparently when you call `.to_html` on the object it (now?) forces the DOCTYPE into an HTML5 doctype. The solution I proposed seemed reasonable to me, at least enough to begin a conversation around it.

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,5 @@ matrix:
   fast_finish: true
 language: ruby
 rvm:
-  - 2.1.9
-  - 2.2.6
   - 2.3.3
   - 2.4.0
```
Comment (grosser) [inline .travis.yml]: we need to add 2.5 / 2.6 soon :D

File: Gemfile.lock [hunk 2]
```diff
@@ -1,78 +0,0 @@
-GIT
```
Comment (grosser) [inline Gemfile.lock]: please keep this file around ... it's the safe-state for this repos development ... https://grosser.it/2015/08/14/check-in-your-gemfile-lock/

File: lib/premailer/premailer.rb [hunk 3]
```diff
@@ -402,7 +402,7 @@ def append_query_string(doc, qs)
 
   # Check for an XHTML doctype
   def is_xhtml?
-    intro = @doc.to_html.strip.split("\n")[0..2].join(' ')
+    intro = @doc.to_xhtml.strip.split("\n")[0..2].join(' ')
```
Comment (grosser) [inline lib/premailer/premailer.rb]: is there a chance to_xhtml breaks in some explosive way when this is simple non-xml text ?

File: Gemfile.lock (hunk 2 above)
Comment (meowsus) [reply Gemfile.lock]: No problem... I was confused by this:

> once this is green we can bring back the rm Gemfile.lock so we actually test with the latest dependencies that end-users would be using :)

It seemed like you were saying that you wanted to see this removal committed in the PR, which I found strange after requesting a bundle update. Now I'm curious what you meant by this :)

File: .travis.yml (hunk 1 above)
Comment (meowsus) [reply .travis.yml]: I'll add them now and bump the existing 2.3 and 2.4 versions to their latest patches while I'm at it. I don't know if anything will be required on the Travis side, though. Admittedly I haven't worked much with Travis, having been shackled to Bamboo for so long...

**Comment by grosser** [comment]
I usually just do `2.3` / `2.4` and let travis auto-bump them :)

**Comment by grosser** [comment]
commit the Gemfile.lock so we have a safe dev env and then `rm
Gemfile.lock` in travis (maybe just a single stage) so we can see if
updating a dependency breaks the build

File: lib/premailer/premailer.rb (hunk 3 above)
Comment (meowsus) [reply lib/premailer/premailer.rb]: It [doesn't seem so](https://github.com/premailer/premailer/pull/381/files#diff-512f50647d2fb1ab8083dcd06b8abeb1R41).

**Comment by grosser** [comment]
awesome!

**Pull request merged**